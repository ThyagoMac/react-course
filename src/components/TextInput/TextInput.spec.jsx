import { render, screen } from "@testing-library/react";
import { TextInput } from ".";
import userEvent from "@testing-library/user-event";

describe("<TextInput />", () => {
  it("Should have a value on searchValue", () => {
    const fn = jest.fn();

    render(
      <TextInput
        handleSearch={fn}
        searchValue={"Testing"}
        placeholder={"Filter posts"}
      />
    );

    const textInput = screen.getByPlaceholderText(/filter posts/i)

    expect(textInput).toBeInTheDocument();
    expect(textInput.value).toBe("Testing");
  });

  it("Should call handleSearch() on each key press", () => {
    const fn = jest.fn();

    render(
      <TextInput
        handleSearch={fn}
        placeholder={"Filter posts"}
      />
    );

    const textInput = screen.getByPlaceholderText(/filter posts/i)
    const val = "user typing text";

    userEvent.type(textInput, val);

    expect(textInput.value).toBe(val);
    expect(fn).toHaveBeenCalledTimes(val.length);
  });

  it("Should match snapshot", () => {
    const fn = jest.fn();

    const { container } = render(
      <TextInput
        handleSearch={fn}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});