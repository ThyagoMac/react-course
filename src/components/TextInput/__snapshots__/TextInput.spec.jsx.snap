// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should match snapshot 1`] = `
<input
  class="text-input"
  type="search"
  value=""
/>
`;
